pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -132,15 +133,29 @@ def _validate_color_args(self):
         else:
             self.color = None
 
+    @cache_readonly
+    def _color_attrs(self):
         # get standard colors for default
-        colors = get_standard_colors(num_colors=3, colormap=self.colormap, color=None)
         # use 2 colors by default, for box/whisker and median
         # flier colors isn't needed here
         # because it can be specified by ``sym`` kw
-        self._boxes_c = colors[0]
-        self._whiskers_c = colors[0]
-        self._medians_c = colors[2]
-        self._caps_c = colors[0]
+        return get_standard_colors(num_colors=3, colormap=self.colormap, color=None)
+
+    @cache_readonly
+    def _boxes_c(self):
+        return self._color_attrs[0]
```

Comment:
I guess we can de-duplicate this in a follow up since 3 variables caches the same `self._color_attrs[0]`?

Reply:
sure.  there will be a few more passes in this less-stateful push